src/remix/sdk-setup.ts

```diff
@@ -0,0 +1,266 @@
+): void {
+  const initCall = builders.functionCall('Sentry.init', {
+    dsn,
+    tracesSampleRate: 1.0,
+    replaysSessionSampleRate: 0.1,
+    replaysOnErrorSampleRate: 1.0,
+    integrations: [
+      builders.newExpression('Sentry.BrowserTracing', {
+        routingInstrumentation: builders.functionCall(
+          'Sentry.remixRouterInstrumentation',
+          builders.raw('useEffect'),
+          builders.raw('useLocation'),
+          builders.raw('useMatches'),
+        ),
+      }),
+      builders.newExpression('Sentry.Replay'),
+    ],
+  });
+
+  const originalHooksModAST = originalHooksMod.$ast as Program;
+  const initCallInsertionIndex = getInitCallInsertionIndex(originalHooksModAST);
+
+  originalHooksModAST.body.splice(
+    initCallInsertionIndex,
+    0,
+    // @ts-expect-error - string works here because the AST is proxified by magicast
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    generateCode(initCall).code,
+  );
+}
+
+function insertServerInitCall(
+  dsn: string,
+  originalHooksMod: ProxifiedModule<any>,
+) {
+  const initCall = builders.functionCall('Sentry.init', {
+    dsn,
+    tracesSampleRate: 1.0,
+  });
+
+  const originalHooksModAST = originalHooksMod.$ast as Program;
+
+  const initCallInsertionIndex = getInitCallInsertionIndex(originalHooksModAST);
+
+  originalHooksModAST.body.splice(
+    initCallInsertionIndex,
+    0,
+    // @ts-expect-error - string works here because the AST is proxified by magicast
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    generateCode(initCall).code,
+  );
+}
+
+export function isRemixV2(
+  remixConfig: PartialRemixConfig,
+  packageJson: PackageDotJson,
+): boolean {
+  const remixVersion = getPackageVersion('@remix-run/react', packageJson);
+  const remixVersionMajor = remixVersion && parse(remixVersion)?.major;
+  const isV2Remix = remixVersionMajor && remixVersionMajor >= 2;
+
+  return isV2Remix || remixConfig?.future?.v2_errorBoundary || false;
+}
+
+export async function loadRemixConfig(): Promise<PartialRemixConfig> {
+  const configFilePath = path.join(process.cwd(), REMIX_CONFIG_FILE);
+
+  try {
+    if (!fs.existsSync(configFilePath)) {
+      return {};
+    }
+
+    const configUrl = url.pathToFileURL(configFilePath).href;
+    const remixConfigModule = (await import(configUrl)) as {
+      default: PartialRemixConfig;
+    };
+
+    return remixConfigModule?.default || {};
+  } catch (e: unknown) {
+    clack.log.error(`Couldn't load ${REMIX_CONFIG_FILE}.`);
+    clack.log.info(
+      chalk.dim(
+        typeof e === 'object' && e != null && 'toString' in e
+          ? e.toString()
+          : typeof e === 'string'
+          ? e
+          : 'Unknown error',
+      ),
+    );
+
+    return {};
+  }
+}
+
+export async function instrumentRootRoute(
+  isV2?: boolean,
+  isTS?: boolean,
+): Promise<void> {
+  const rootFilename = `root.${isTS ? 'tsx' : 'jsx'}`;
```

Comment:
Do you think it's likely that people mix up ts and js? Just wondering if we should make a lookup regardless of `isTS`, similarly to what we do with `findScriptFile` in the SvelteKit wizard. If this is unlikely or even not possible in remix then let's leave it as is. I don't think it's a blocker either way.

Reply:
It's possible, but I don't think it's very likely for the files we are modifying here.

Reply:
Okay, then let's leave it as-is for now and revisit if it comes up.